api.py

```diff
@@ -163,11 +215,20 @@ def scan(
             vulns = {}
             for port in nm[host]["tcp"]:
                 product = nm[host]["tcp"][port]["product"]
-                Vulnerablities = self.SearchVuln(nm[host]["tcp"][port], apiKey, debug)
+                Vulnerablities = self.SearchVuln(nm[host]["tcp"][port], vuln_api_key, shodan_api_key, zoomeye_api_key, debug)
                 if Vulnerablities:
-                    vulns[product] = Vulnerablities
+                    for vuln_id, vuln_info in Vulnerablities.items():
+                        vulns[vuln_id] = vuln_info
 
             self.scan_results[host]["vulns"] = vulns
+            #add all vulnerabilities to the list
+            all_vulnerabilities.extend(vulns.values())
+
+        # Exploit the vulnerabilities found
+        if all_vulnerabilities:
+            GetExploitsFromArray(all_vulnerabilities, log, console, max_exploits=max_exploits)
+            if scan_vulns:
+                exploit_vulnerabilities(all_vulnerabilities, target, log, console, max_exploits=max_exploits)
 
```

Comment:
The expoits should only be exploited if user gives --exploit argument to the program.